parachain/primitives/runtime-common/src/lib.rs

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+//! # Runtime Common
+//!
+//! Common traits and types shared by runtimes.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use bp_rococo::AccountId;
+use core::marker::PhantomData;
+use frame_support::traits::Get;
+use snowbridge_core::{outbound::SendMessageFeeProvider, sibling_sovereign_account_raw};
+use xcm::prelude::*;
+use xcm_builder::{deposit_or_burn_fee, HandleFee};
+use xcm_executor::traits::{FeeReason, TransactAsset};
+
+/// A `HandleFee` implementation that takes fees from `ExportMessage` XCM instructions
+/// to Snowbridge and splits off the local remote fee and deposits it to the origin
+/// parachain sovereign account. The local fee is then returned back to be handled by
+/// the next fee handler in the chain. Most likely the treasury account.
+pub struct XcmExportFeeToSnowbridge<TokenLocation, EthereumNetwork, AssetTransactor, OutboundQueue>(
```

Comment:
Just a general question - How is this `XcmExportFeeToSnowbridge` type intended to be wired up in the runtime, assuming ParityBridge also has a fee handler which it needs to install.

I'm assuming there will be some kind of tuple or something:
```rust
FeeManager = (XcmExportFeeToParityBridge, XcmExportFeeToSnowbridge)
```

I guess I'm just worried that the two fee handlers will conflict in some manner.

Reply:
Actually it's `type FeeManager = (XcmExportFeeToSnowbridge,XcmExportFeeToParityBridge)`
 
The `HandleFee` impl for Tuple as following
https://github.com/Snowfork/polkadot-sdk/blob/f56aaf93e36117392e61e48a59e66ae764d697ff/polkadot/xcm/xcm-builder/src/fee_handling.rs#L40-L56

so remote fee to assethub sovereign and the leftover(i.e. local fee) to treasury.